service/gcs/runtime/runc/runc.go

```diff
@@ -383,39 +383,41 @@ func (r *runcRuntime) pidMapToProcessStates(pidMap map[int]*runtime.ContainerPro
 	return processStates
 }
 
-// waitOnProcess waits for the process to exit, and returns its
-// oslayer.ProcessExitState containing exit information.
-// TODO: We might want to give more options for this, such as specifying WNOHANG.
-func (r *runcRuntime) waitOnProcess(pid int) (oslayer.ProcessExitState, error) {
+// waitOnProcess waits for the process to exit, and returns its exit code.
+//
+// TODO: We might want to give more options for this, such as specifying
+// WNOHANG.
+func (r *runcRuntime) waitOnProcess(pid int) (int, error) {
 	process, err := os.FindProcess(pid)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to find process %d", pid)
+		return -1, errors.Wrapf(err, "failed to find process %d", pid)
 	}
 	state, err := process.Wait()
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed waiting on process %d", pid)
+		return -1, errors.Wrapf(err, "failed waiting on process %d", pid)
 	}
-	return realos.NewProcessExitState(state), nil
+
+	return state.Sys().(syscall.WaitStatus).ExitStatus(), nil
```

Comment:
In recent Go versions, they added a function to get the exit code directly without this cast. It might be Go 1.12, though, so maybe we can't use it yet.

Reply:
Yea its in 1.12